### PR TITLE
Fixed bug in admin/categories

### PR DESCRIPTION
**Pull request by Dreedle**
-In admin/categories controller, new_or_edit action (line 27), added conditionals to determine definition of @category depending on params[:action]. This resolved problems with category creation and editing.

-Added rspec test for adding new category. (categories_controller_spec, line 37)

-Added cucumber tests for adding and editing categories (features/create_category.feature, features/edit_category.feature)

### CODE REVIEW

File: app/controllers/admin/categories_controller.rb [hunk 1]
```diff
@@ -24,13 +24,21 @@ def destroy
   private
 
   def new_or_edit
+    if params[:action] == "new"
+      @category = Category.new
+    elsif params[:action] == "edit" && params[:id]
+      @category = Category.find(params[:id])
+      @category.attributes = params[:category]
+    elsif params[:action] == "edit" && !params[:id]
```
Comment (kariabancroft) [inline app/controllers/admin/categories_controller.rb:32]: What would the scenario be where this could happen?

Comment (Dreedle) [inline app/controllers/admin/categories_controller.rb:32]: The first "if" statement is for when you first navigate to the page by clicking the "Categories" link (admin/categories/new). The first "elsif" is for when you navigate to to "/admin/categories/edit/:id" by clicking the "edit" link for a category. The second "elsif" is to handle the post request (which uses the "edit" action) when the category is new (which is why there wouldn't be an id in the params).

Comment (kariabancroft) [inline app/controllers/admin/categories_controller.rb:32]: Got it! Makes sense!

**Comment by kariabancroft** [comment]
Ok, looks good to close

**Pull request closed without merging**